contracts/implementation/PoolFactory.sol

```diff
@@ -136,6 +143,13 @@ contract PoolFactory is IPoolFactory, ITwoStepGovernance {
             "Decimal precision too high"
         );
 
+        if (deploymentFeeToken != address(0) && deploymentFee != 0) {
```

Comment:
its more readable but not necessary to have both checks. In the initial case deploymentFeeToken and deploymentFee will both be 0, but you cant change deploymentFee unless deploymentFeeToken not the 0 address, once its set you can never set it back to the 0 address, you can only ever set deploymentFee back to 0